internal/auth/session_middleware.go

```diff
@@ -47,29 +47,15 @@ func cookieAuth(ctx context.Context, oryApiClient *client.APIClient, oryKratosSe
 	return session.Identity.Id, nil
 }
 
-func tokenAuth(tokenRepository tokenRepository, header string) (string, error) {
-	// get the user id from the database.
-	// check if we need to strip a bearer prefix
-	if len(header) > 7 && header[:7] == "Bearer " {
-		header = header[7:]
-	}
-	return tokenRepository.GetUserIDByToken(header)
-}
-
-func SessionMiddleware(oryApiClient *client.APIClient, tokenRepository tokenRepository) echo.MiddlewareFunc {
+func SessionMiddleware(oryApiClient *client.APIClient, verifier verifier) echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) (err error) {
 			oryKratosSessionCookie := getCookie("ory_kratos_session", c.Cookies())
 
 			var userID string
 
 			if oryKratosSessionCookie == nil {
-				// check for authorization header
-				authorizationHeader := c.Request().Header.Get("Authorization")
-				if authorizationHeader == "" {
-					return c.JSON(401, map[string]string{"error": "no session, missing authorization header"})
-				}
-				userID, err = tokenAuth(tokenRepository, authorizationHeader)
+				userID, err = verifier.VerifyCryptoChallenge(c.Request())
```

Comment:
This can be renamed as well to make it simpler to understand what is really going on. `VerifyRequestSignature`